compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantValueResolver.java

```diff
@@ -494,4 +510,53 @@ private void checkUniqueness(BLangConstant constant) {
             }
         }
     }
+
+    private void updateSymbolType(BLangConstant constant) {
+        BConstantSymbol symbol = constant.symbol;
+        if (symbol.type.getKind() != TypeKind.FINITE && symbol.value != null) {
```

Comment:
Do we still need this check? Isn't a constant only visited once now? Even if not, is it possible for something that was visited previously to resolve to a finite type only in subsequent visits?

Reply:
For example `const int X = 1;`, symbol type is already set to finite type by the `SymbolEnter`. Therefore, IMO we still need this check in here to avoid unnecessary type updates.